File: MTGSearch/src/main/java/com/dbottillo/mtgsearchfree/view/views/MTGCardImageView.java
package com.dbottillo.mtgsearchfree.view.views;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.ImageView;

public class MTGCardImageView extends ImageView {
    public MTGCardImageView(Context context) {
        super(context);
    }

    public MTGCardImageView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public MTGCardImageView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(widthMeasureSpec, (int) (widthMeasureSpec / MTGCardView.RATIO_CARD));
    }

}

File: core/src/main/kotlin/com/dbottillo/mtgsearchfree/ui/views/MTGCardImageView.kt
package com.dbottillo.mtgsearchfree.ui.views

import android.content.Context
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatImageView
import com.dbottillo.mtgsearchfree.Constants

class MTGCardImageView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : AppCompatImageView(context, attrs, defStyleAttr) {

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
        val height = (measuredWidth * Constants.RATIO_CARD).toInt()
        setMeasuredDimension(measuredWidth, height)
    }
}